vgscollect/src/main/java/com/verygoodsecurity/vgscollect/view/internal/CVCInputField.kt

```diff
@@ -1,24 +1,50 @@
 package com.verygoodsecurity.vgscollect.view.internal
 
 import android.content.Context
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
 import android.os.Build
 import android.text.InputFilter
 import android.text.InputType
 import android.view.Gravity
 import android.view.View
+import androidx.core.content.ContextCompat
+import com.verygoodsecurity.vgscollect.R
 import com.verygoodsecurity.vgscollect.core.model.state.Dependency
 import com.verygoodsecurity.vgscollect.core.model.state.FieldContent
 import com.verygoodsecurity.vgscollect.core.storage.DependencyType
+import com.verygoodsecurity.vgscollect.view.card.CardType
 import com.verygoodsecurity.vgscollect.view.card.FieldType
 import com.verygoodsecurity.vgscollect.view.card.conection.InputCardCVCConnection
 import com.verygoodsecurity.vgscollect.view.card.text.CVCValidateFilter
 import com.verygoodsecurity.vgscollect.view.card.validation.CardCVCCodeValidator
+import com.verygoodsecurity.vgscollect.view.internal.CVCInputField.PreviewIconGravity.END
+import com.verygoodsecurity.vgscollect.view.internal.CVCInputField.PreviewIconGravity.START
+import com.verygoodsecurity.vgscollect.view.internal.CVCInputField.PreviewIconVisibility.*
 
 /** @suppress */
-internal class CVCInputField(context: Context): BaseInputField(context) {
+internal class CVCInputField(context: Context) : BaseInputField(context) {
 
     override var fieldType: FieldType = FieldType.CVC
-    private var cvcLength:Array<Int> = arrayOf(3,4)
+    private var cvcLength: Array<Int> = arrayOf(3, 4)
+    private var cardType: CardType = CardType.UNKNOWN
+
+    private val iconBounds = Rect(
+        0,
+        0,
+        context.resources.getDimension(R.dimen.c_icon_size_w).toInt(),
+        context.resources.getDimension(R.dimen.c_icon_size_h).toInt()
+    )
+
+    private var defaultPreviewIcon = ContextCompat.getDrawable(context, R.drawable.ic_card_front_preview_dark).also {
+            it?.bounds = iconBounds
+    }
+    private var amExPreviewIcon = ContextCompat.getDrawable(context, R.drawable.ic_card_front_preview_dark_4).also {
+        it?.bounds = iconBounds
+    }
```

Comment:
It is possible in the future that we will have more cases with brand icons. I think there is no sense to store all of them. Furthermore, it becomes redundant if a developer doesn't use the preview icon.
Could we move this part of code and `getIcon(_)` method to the class\extension\helper and generate an icon only on demand?

Reply:
@Endorf This is basic implementation, it will change in next PR's.

Reply:
@Endorf Also about redundancy, I think as we have only 2 drawable, it's less redundant than getting it each time when text/etc. changes, most native android view storing links to drawables/paints, etc.  For changes in the future, we have CardIconAdapter, where dev can set any icon he wants for cardType/cvcLength

Reply:
@Endorf Best option here will be by lazy